p24/src/main.rs

```diff
@@ -0,0 +1,50 @@
+fn main() {
+    let arr = [2, 4, 8, 16];
+
+    let mut n = 2;
+    let nth = nth_item(&arr, &n);
+    let increased = increased_by_first_item(&arr, &mut n);
+
+    let value = {
+        let values = TwoValues::new(&arr[3], increased);
+
+        assert_eq!(*values.get_first(), 16);
+
+        values.get_second()
+    };
+
+    assert_eq!(*value, 4);
+    assert_eq!(*nth, 8);
+}
+
+// +++ Added lifetime 'a to function and return type
+fn nth_item<'a>(data: &'a [usize], n: &usize) -> &'a usize {
+    &data[*n]
+}
+
+// +++ Added lifetime 'a to function and return type
+fn increased_by_first_item<'a>(data: &[usize], n: &'a mut usize) -> &'a mut usize {
+    *n += data[0];
+    n
+}
+
+// +++ Added lifetime to struct
+struct TwoValues<'a> {
```

Comment:
Note that the struct could have two different lifetimes, i.e. `TwoValues<'a, 'b>`, but this also works.